### PR TITLE
Traefik 2.0 Configuration 

### PR DESCRIPTION
**Pull request by OscarKevinXR**
* Updated the `examples/host-vars.yml` file to have the `matrix_postgres_connection_password` be presented upfront similar to the matrix-docker-ansible-deploy repo. 
* Added  `nextcloud_nextcloud_docker_container_extra_arguments` to `roles/nextcloud-server/defaults/main.yml` and `roles/nextcloud-server/templates/systemd/nextcloud-apache.service.j2` 
* Replicated the documentation for `configuring-playbook-own-webserver.md` and adjusted for the nextcloud variables 

I also proposed small documentation updates to add navigation to this new information.

### CODE REVIEW

File: roles/nextcloud-server/defaults/main.yml [hunk 1]
```diff
@@ -75,10 +78,10 @@ nextcloud_nextcloud_apache_container_http_host_bind_port: "{{ '' if nextcloud_ng
 # on this port by default.
 nextcloud_apache_http_port_on_host: 37150
 
-nextcloud_postgres_connection_hostname: "nextcloud-postgres"
 nextcloud_postgres_connection_username: "nextcloud"
 nextcloud_postgres_connection_password: "nextcloud-password"
 nextcloud_postgres_db_name: "nextcloud"
+nextcloud_postgres_connection_hostname: "nextcloud-postgres"
```
Comment (OscarKevinXR) [inline roles/nextcloud-server/defaults/main.yml:84]: I updated the order of the variables here to reflect the order required for initializing a new nextcloud instance through the GUI once successfully setup and deployed.

File: docs/configuring-playbook-own-webserver.md [hunk 2]
```diff
@@ -26,4 +35,95 @@ nextcloud_nginx_proxy_enabled: false
...
+  - '--label "traefik.enable=true"'
+
+  # The Nginx proxy container will receive traffic from these subdomains
+  - '--label "traefik.http.routers.nextcloud-nginx-proxy.rule=Host(`{{ host_specific_nextcloud_hostname }}`)"'
+
+  # (The 'web-secure' entrypoint must bind to port 443 in Traefik config)
+  - '--label "traefik.http.routers.nextcloud-nginx-proxy.entrypoints=web-secure"'
+
+  # (The 'default' certificate resolver must be defined in Traefik config)
+  - '--label "traefik.http.routers.nextcloud-nginx-proxy.tls.certResolver=default"'
```
Comment (spantaleev) [inline docs/configuring-playbook-own-webserver.md:73]: Even with this, `nextcloud-nginx-proxy` would sitll be installed and would still try to occupy ports `80` and `443`, won't it? We'd need something else added here (and to the playbook) to disable it.

Comment (OscarKevinXR) [reply docs/configuring-playbook-own-webserver.md:73]: That's a good point! I'll take a look and revisit this PR with changes that reflect this suggestion.

**Comment by OscarKevinXR** [comment]
The latest commit should disable port 80 from being occupied on the host device by setting `nextcloud_nextcloud_apache_container_http_host_bind_port: ''` which disables the port from being set in the [nextcloud-apache.service.j2](https://github.com/spantaleev/nextcloud-docker-ansible-deploy/blob/10078395a53b99fda723023fc7940b66f5691951/roles/nextcloud-server/templates/systemd/nextcloud-apache.service.j2#L38-L40) file. 
 
Screenshot from 2022-09-18 15-41-25

**Comment by spantaleev** [comment]
Looks good! Thanks!

**Pull request merged**